packages/adapter-fauna/src/index.ts

```diff
@@ -0,0 +1,249 @@
+import { LuciaError, generateRandomString } from "lucia-auth";
+import type { Adapter } from "lucia-auth";
+import faunadb from "faunadb";
+import {convertUserResponse} from "./utils.js";
+const {query, Client, errors} = faunadb;
+const {FaunaError} = errors;
+const q = query;
+
+const adapter = (
+	// @ts-ignore
+	faunaClient: Client,
+	// @ts-ignore
+	errorHandler: (error: FaunaError) => void = () => {}
+): Adapter => {
+	return {
+		getUser: async (userId) => {
+			try {
+				return await faunaClient.query(
```

Comment:
Preferably, promise should be handled by `async/await` rather than callbacks.

```ts
const response = await faunaClient.query();
const users = res.data;
return users.length !== 0 ? users[0].data : null;
```

Reply:
Refactored. Out of curiosity, Is this a personal preference or a performance benefit?